src/shared/components/Buttons.js

```diff
@@ -0,0 +1,120 @@
+import classnames from 'classnames';
+
+import { SvgIcon } from '@hypothesis/frontend-shared';
+
+/**
+ * @typedef ButtonProps
+ * @prop {import("preact").ComponentChildren} [children]
+ * @prop {string} [className]
+ * @prop {string} [icon] - name of `SvgIcon` to render in the button
+ * @prop {'left'|'right'} [iconPosition] - Icon positioned to left or to
+ *   right of button text
+ * @prop {boolean} [disabled]
+ * @prop {boolean} [expanded] - Is the element associated with this button
+ *   expanded (set `aria-expanded`)
+ * @prop {boolean} [pressed] - Is this button currently "active?" (set
+ *   `aria-pressed`)
+ * @prop {() => any} [onClick]
+ * @prop {string} [title] - Button title; used for `aria-label` attribute
+ * @prop {'primary'} [variant] - For styling: element variant
+ */
+
+/**
+ * @typedef IconButtonBaseProps
+ * @prop {string} icon - Icon is required for icon buttons
+ * @prop {string} title - Title is required for icon buttons
+ */
+
+/**
+ * @typedef {ButtonProps & IconButtonBaseProps} IconButtonProps
+ */
+
+/**
+ * @param {ButtonProps} props
+ */
+function ButtonBase({
+  children,
+  className,
+  icon,
+  iconPosition = 'left',
+  disabled,
+  expanded,
+  pressed,
+  onClick = () => {},
+  title,
+  variant,
+}) {
+  const ariaAttributes = {};
+  const otherAttributes = {};
+  if (typeof disabled === 'boolean') {
+    otherAttributes.disabled = disabled;
+  }
+  if (typeof title !== 'undefined') {
+    otherAttributes.title = title;
+    ariaAttributes['aria-label'] = title;
+  }
+
+  if (typeof expanded === 'boolean') {
+    ariaAttributes['aria-expanded'] = expanded;
+  }
+  if (typeof pressed === 'boolean') {
+    ariaAttributes['aria-pressed'] = pressed;
+  }
+
+  return (
+    <button
+      className={classnames(className, {
+        [`${className}--${variant}`]: variant,
+        [`${className}--icon-${iconPosition}`]: icon,
+      })}
+      onClick={onClick}
+      {...otherAttributes}
+      {...ariaAttributes}
+    >
+      {children}
+    </button>
+  );
+}
+
+/**
+ * An icon-only button
+ * @param {IconButtonProps} props
```

Comment:
```suggestion
 * An icon-only button
 *
 * @param {IconButtonProps} props
```

Our usual convention is to leave a new line between the summary and params.